cli/internal/plugins/git/versioner.go

```diff
@@ -0,0 +1,72 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/buffalo-cli/plugins"
+)
+
+type Versioner struct {
+	pluginsFn plugins.PluginFeeder
+}
+
+func (b *Versioner) WithPlugins(f plugins.PluginFeeder) {
+	b.pluginsFn = f
+}
+
+func (b *Versioner) ScopedPlugins() []plugins.Plugin {
+	if b.pluginsFn == nil {
+		return []plugins.Plugin{}
+	}
+
+	var scoped []plugins.Plugin
+	for _, p := range b.pluginsFn() {
+		switch p.(type) {
+		case VersionRunner:
```

Comment:
I think this is the part I still need to master, the `ScopedPluggins` and `Plugins` methods. Thanks for adding this example, I will re-read it soon.